server/game/cards/08-MotC/AkodoMakoto.js

```diff
@@ -0,0 +1,37 @@
+const DrawCard = require('../../drawcard.js');
+const { Players, CardTypes } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class AkodoMakoto extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Remove fate/discard character',
+            when: {
+                afterConflict: (event, context) => {
+                    return event.conflict.winner === context.player && context.source.isParticipating();
+                }
+            },
+            target: {
+                cardType: CardTypes.Character,
+                controller: Players.Any,
+                cardCondition: (card, context) => {
+                    return card.hasTrait('courtier') && card.isParticipating() &&
+                        (card.fate > 0 ? card.allowGameAction('removeFate', context) : card.allowGameAction('discardFromPlay', context));
+                }
+            },
+            effect: '{1} {0}',
+            effectArgs: context => context.target.fate > 0 ? 'remove 1 fate from' : 'discard',
+            handler: context => {
```

Comment:
I realise you're probably just copying from an older card implementation, but I'd like to avoid `handler` where possible for new cards.

For this I think `gameAction: context => context.target.fate > 0 ? AbilityDsl.actions.removeFate : AbilityDsl.actions.discardFromPlay()` should mean that you can simplify `cardCondition`, and you then don't need `effect`, `effectArgs` or `handler`

Reply:
Would this not be the ideal situation for your `otherwise` action?

Reply:
yeah I was just copying deathseeker I didn't know there was an issue with handler. I'll fix it

Reply:
`otherwise` is harder than I first thought, and I think it's not a great name for it.  The action I was trying to write was for 'if able' effects (and I think that might be a better name).  Here, I think we want a conditional game action.  I'll try and write something.